dao/src/main/java/com/epam/esm/dao/impl/CertificateDaoImpl.java

```diff
@@ -0,0 +1,126 @@
+package com.epam.esm.dao.impl;
+
+import com.epam.esm.dao.CertificateDao;
+import com.epam.esm.dao.entity.Certificate;
+import com.epam.esm.dao.entity.Tag;
+import com.epam.esm.dao.exception.NoSuchResultException;
+import com.epam.esm.dao.impl.sorting.CertificateSortBy;
+import com.epam.esm.dao.impl.sorting.SortParameterInserter;
+import com.epam.esm.dao.impl.sorting.SortingParameter;
+import com.epam.esm.dao.impl.util.CertificateSearchCriteria;
+import com.epam.esm.dao.impl.util.SortingCalculator;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.domain.EntityScan;
+import org.springframework.stereotype.Repository;
+
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+import javax.persistence.criteria.*;
+import java.util.ArrayList;
+import java.util.List;
+
+@Repository
+@EntityScan(basePackages = "com.epam.esm.dao.entity")
```

Comment:
Could we move `EnitytScan` annotation to the start application class level (`@SpringBootApplication` class)